src/WinRT.Runtime/TypeNameSupport.cs

```diff
@@ -473,6 +473,16 @@ private static bool TryAppendWinRTInterfaceNameForType(Type type, StringBuilder
                 return true;
             }
 
+#if NET8_0_OR_GREATER

```

Comment:
What scenario will actually hit this?

Reply:
It was mostly for consistency. I was thinking perhaps there could be some weird case where you could manage to end up calling the name resolution logic on such a type, so might as well check and throw here too? 🤔

Reply:
This should only be called for GetRuntimeClassName.  That shouldn't be possible given we won't marshal it.